alphadia/workflow/peptidecentric/extraction_handler.py

```diff
@@ -0,0 +1,140 @@
+import pandas as pd
+import seaborn as sns
+from alphabase.spectral_library.base import SpecLibBase
+
+from alphadia.data.alpharaw_wrapper import AlphaRaw
+from alphadia.data.bruker import TimsTOFTranspose
+from alphadia.peakgroup import search
+from alphadia.peakgroup.config_df import HybridCandidateConfig
+from alphadia.plexscoring.config import CandidateConfig
+from alphadia.plexscoring.plexscoring import CandidateScoring
+from alphadia.reporting.reporting import Pipeline
+from alphadia.workflow.config import Config
+from alphadia.workflow.managers.optimization_manager import OptimizationManager
+
+
+class ExtractionHandler:
+    """Manages precursor and fragment extraction operations."""
+
+    def __init__(
+        self,
+        config: Config,
+        optimization_manager: OptimizationManager,
+        reporter: Pipeline,
+        spectral_library: SpecLibBase,
+        *,
+        rt_column: str,
+        mobility_column: str,
+        precursor_mz_column: str,
+        fragment_mz_column: str,
+    ):
+        self._config: Config = config
+        # TODO think about passing only what is needed, e.g. rt_error, ms1_error, etc.
```

Comment:
I'm not sure if this is a good pattern. I would rather encapsulate the state in context objects.